docs/src/contributing/development-environment.md

```diff
@@ -1,280 +1,278 @@
 
-!!! note
-    Previously, Deis used boot2docker to run the registry. However, Docker has deprecated boot2docker in favor of Docker Machine.
+Install Docker Machine according to the normal installation instructions, then use it to create a new VM:
 
-Install Docker Machine according to the normal installation instructions. Then create a new image for hosting your Deis Docker registry:
+```
+$ docker-machine create deis-docker \
+    --driver=virtualbox \
+    --virtualbox-disk-size=100000 \
+    --engine-insecure-registry=192.168.0.0/16
+```
 
-    $ docker-machine create --driver virtualbox --virtualbox-disk-size=100000 \
-      --engine-insecure-registry=192.168.0.0/16 deis-registry
+This will create a new virtual machine named `deis-docker` that will take up as much as 100,000 MB of disk space. The images you build may be large, so allocating a big disk is a good idea.
 
-This will create a new virtual machine named `deis-registry` that will take up as much as 100,000 MB of disk space. Registries tend to be large, so allocating a big disk is a good idea.
+Once the `deis-docker` machine exists, source its values into your environment so your docker client knows how to use the new machine. You may even choose to add this to your bash profile or similar.
 
-Once the deis-registry machine exists, source its values into your environment so your docker client knows how to use the new machine.
+```
+$ eval "$(docker-machine env docker-deis)"
+```
 
-    $ eval "$(docker-machine env deis-registry)"
+After following these steps, some Docker Machine users report a slight delay (30 - 60 seconds) before the Docker server is ready.
 
 !!! note
-    Because the registry that we create will not have a valid SSL certificate, we run the local registry as an insecure (HTTP, not HTTPS) registry. Each time Docker Machine reboots, the registry will get a new IP address somewhere in the `192.168.0.0/16` range. We must declare that explicitly when configuring Docker Machine.
-
-At this point, our `deis-registry` VM can now serve as a registry for Deis' Docker images. Later we will return to this.
+    In subsequent steps, you may run a Docker registry within the `deis-docker` VM. Such a registry will not have a valid SSL certificate and will use HTTP instead of HTTPS. Such registries are implicitly untrusted by the Docker server (which is also running on the `deis-docker` VM).  In order for the Docker server to trust the insecure registry, `deis-docker` is explicitly created to trust all registries in the `192.168.0.0/16` range - the range in which the VM (and therefore the registry) will exist.  This will effectively permit Docker pulls and pushes to such a registry.
```

Comment:
I would just add a note/caveat that this range may not be the one used in all cases, and to confirm with the machine's IP.